### PR TITLE
examples/rpl_udp: Fix unconditional setting of a radio channel on initialization

### PR DESCRIPTION
**Pull request by BytesGalore**
The radio channel is set unconditionally on initialization of a node, it is set to `RADIO_CHANNEL` which is out of range e.g. for the AT86RF23 on the iot-lab_M3.
Also it overrides the setting if the channel has been set previously, i.e. using the `chan` command before invoking `init (r|n|h)`.

This PR changes the `init` procedure to use the set or default channel provided by the `TRANSCEIVER` and only tries to set it to `RADIO_CHANNEL` if the transceiver cannot use the set channel.
On failing setting the channel the `init` procedure aborts.

### CODE REVIEW

**Comment by gebart** [comment]
Should we change the header `rpl_udp.h` to use `#ifndef` for the `RADIO_CHANNEL` macro etc. so that is possible to override it while building?

**Comment by BytesGalore** [comment]
@gebart something like `make BOARD=iot-lab_M3 RADIO_CHANNEL=17` right?
Hmm, maybe we should then make the radio channel setup as compile parameter independent from this example (this will look more difficult if have more than one transceiver mounted on a board)?

**Comment by LudwigOrtmann** [comment]
How about defining a default channel per radio driver?

**Comment by BytesGalore** [comment]
@gebart @LudwigOrtmann another option could be to set a valid default value for the device channel on `netdev_t::init()`

**Comment by LudwigOrtmann** [comment]
I'd leave that to the application.

**Comment by LudwigOrtmann** [comment]
(But I don't have a strong opinion here, maybe @thomaseichinger or @rousselk want to comment?)

**Comment by thomaseichinger** [comment]
> I'd leave that to the application.

:+1: 
It is hard to define a default channel as IEEE 802.15.4 defines channels 0-10 for sub-GHz and 11-26 in GHz band.

**Comment by cgundogan** [comment]
Without digging in any code - actually, having the device driver knowing about its "default channel" would be great for the *wire-and-forget* use case, where I do not want to think about channels for same devices. Since the channels can vary depending on the underlying physical layer / medium access strategy, this information would end up in their device drivers?

However, @BytesGalore , the diff looks good to me. But I want to await @OlegHahm 's comment on this.

**Comment by LudwigOrtmann** [comment]
@cgundogan What @thomaseichinger pointed out: The same driver may be used with different frequency bands. So a default channel can only be defined if you know how the radio is actually being used. We could add this information to boards. But even so the same board may be available in different configurations - the chronos for example is available for different bands.

**Comment by thomaseichinger** [comment]
I think the passiv approach proposed in this PR provides more flexibility than using predefined generic default channel (especially at a upper layer which doesn't care about the PHY below). Each radio device configures itself to a default channel after reset. (Fingers crossed it is the same for each device of the same model).

**Comment by BytesGalore** [comment]
@thomaseichinger at least the [at86rf231](http://www.atmel.com/images/doc8111.pdf) does.

**Comment by OlegHahm** [comment]
I agree with @thomaseichinger. This application is also used by various by people for simple experiments where it is very desirable to have the channel configurable.

File: examples/rpl_udp/rpl.c [hunk 1]
```diff
@@ -44,7 +44,7 @@ void rpl_udp_init(int argc, char **argv)
 {
     transceiver_command_t tcmd;
     msg_t m;
-    uint32_t chan = RADIO_CHANNEL;
+    uint32_t chan = 0;
```
Comment (OlegHahm) [inline examples/rpl_udp/rpl.c]: Shouldn't this be a signed value?

Comment (BytesGalore) [inline examples/rpl_udp/rpl.c]: yup

**Comment by BytesGalore** [comment]
@OlegHahm so you suggest to leave everything as it is right?

@gebart I played around with compile time setting the radio channel like suggested (`make RADIO_CHANNEL=13`).
I found it hard to decide when to prefer `RADIO_CHANNEL` over the set channel from within the application.
For instance if you set a channel in the application before calling `init r`.

**Comment by OlegHahm** [comment]
@BytesGalore, correct! :+1:

**Comment by cgundogan** [comment]
So the consensus is to leave this PR as is? I can count 3 *ACKs* here ( @OlegHahm @thomaseichinger and me) and 0 *NACKs*. When travis gets happy, IMHO this can be merged.

**Comment by BytesGalore** [comment]
@cgundogan I kicked T.

File: examples/rpl_udp/rpl.c [hunk 2]
```diff
@@ -70,6 +70,28 @@ void rpl_udp_init(int argc, char **argv)
...
+        }
+        
+        m.type = SET_CHANNEL;
+        msg_send_receive(&m, &m, transceiver_pid);
+        if( chan == 0 ) {
+            puts("ERROR: channel NOT set! Aborting initialization.");
+            return;
+        }
+
+        printf("Channel set to %u\n", (unsigned int)chan);
```
Comment (cgundogan) [inline examples/rpl_udp/rpl.c]: Since chan is defined as int32_t, keep chan signed when printing

Comment (BytesGalore) [inline examples/rpl_udp/rpl.c]: good point, changed.

**Comment by mehlis** [comment]
travis is happy: merge

**Pull request merged**